### DISCUSSION TITLE
feat: I would like to have some Chart components

### DISCUSSION THREAD

**Discussion by KrishnaKumar-1409** [microsoft/fluentui-blazor#1221] [Ideas]
# 🙋 Feature Request
Other Blazor Libraries have Chart Feature , Only Fast Blazor don't have it

**Comment by vnbaaij** [comment]
We will keep this open for a while so people can upvote this.

Also, a bit more text and context on what exactly you would like to see us implement would be helpful 😉

**Comment by Narvalex** [comment]
Implementing the [charts of Fluent UI React](https://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/heads/master/charting/dist/index.html#/examples/areachart) in Fluent UI Blazor would be nice

**Comment by KrishnaKumar-1409** [comment]
Actually I Use Blazor

**Comment by michaelc06** [comment]
Currently using: Blazor-ApexCharts

**Comment by codeuh** [comment]
Charts would be a nice addition to this component library! 

Like @michaelc06, I’m currently using Blazor-ApexCharts alongside fluentui-blazor. I’m achieving great results, but I’m only designing around a single theme. It would become much more cumbersome to support multi-theme sites with this combination. 

Having chart components that use the design token system provided by fluentui-blazor would be very interesting. As for what I’d like to see… At least the chart options provided by the other Microsoft library @Narvalex shared.

**Comment by mahald** [comment]
It would be greatly beneficial to have a wrapper for chartjs.org. This is because it offers a wide variety of charts, including multi-axis charts which I find myself using quite frequently.
https://www.chartjs.org/
https://www.chartjs.org/docs/latest/samples/other-charts/scatter-multi-axis.html

**Comment by KrishnaKumar-1409** [comment]
Yes ,This is needed in blazor fast design components

**Comment by vnbaaij** [comment]
> It would be greatly beneficial to have a wrapper for chartjs.org. This is because it offers a wide variety of charts, including multi-axis charts which I find myself using quite frequently. https://www.chartjs.org/ https://www.chartjs.org/docs/latest/samples/other-charts/scatter-multi-axis.html

I'm not sure yet if we will be adding Chart capabilities, but I do know that if we are going to do it, it is not just going to be a wrapper of a js lib.

**Comment by mahald** [comment]
> > It would be greatly beneficial to have a wrapper for chartjs.org. This is because it offers a wide variety of charts, including multi-axis charts which I find myself using quite frequently. https://www.chartjs.org/ https://www.chartjs.org/docs/latest/samples/other-charts/scatter-multi-axis.html
> 
> I'm not sure yet if we will be adding Chart capabilities, but I do know that if we are going to do it, it is not just going to be a wrapper of a js lib.

I hope that when chart support is introduced, it will include the ability to handle multi-axis charts.

**Comment by sardar97** [comment]
when this would be available for fluent ui blazor??

**Comment by sardar97** [comment]
Are there any chance to see chart js with Fluent Blazor?

**Reply by vnbaaij** [reply]
Not in the near future, no.

**Comment by AJ1000** [comment]
Try out https://puresourcecode.com/dotnet/blazor/blazor-component-for-chartjs/

**Reply by sardar97** [reply]
thank u very much

**Comment by brysageek** [comment]
Wouldn't it be more beneficial to not include Charting??  There are many charting libs out there and they can easily be used with Blazor and Blazor Fluent UI wouldn't we want this community to focus on Fluent UI in Blazor and not charting?  i.e. the community and contributors can focus on bugs/issues/Fluent UI 2 then charting.

**Reply by sardar97** [reply]
what about fluent design ? are there any chart lib with fluent design and use same theme which fluent ui has? like dark mode or colours ...etc

**Reply by vnbaaij** [reply]
We are just discussing setting up a new 'FluentUI Blazor Community' GitHub organization in #2419.

I think a charting project and package could very well be created there. Created by and for the community. 

I would start by creating Blazor versions of the available Fluent UI React charts mentioned earlier. 

We will try to support such a community effort in the ways we can, but this will not be something we will add to the official source and packages. We just do not have the capacity to do that.

**Reply by brysageek** [reply]
That's a good question, and a good point.  I do see the Fluent UI 2 shows examples of some basic charts, but Fluent is more then just a UX Framework right?? Its also a set of guidelines.  Seeing this here https://microsoft.github.io/fluentui-charting-contrib/docs/Detailed%20Implementation%20Steps, and I do like the above suggestions for a wrapper for chartjs.  I have seen talks of an extensions library... Hmmm is there a Feature Request in for this?  Didn't think I had seen one.  We use chart.js at my company too and we recently adopted Blazor Fluent UI for all of web apps, if the were a request that a contributor could grab might be able to make a wrapper possible.
